src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -209,6 +212,72 @@ private static async Task<Document> InlineTemporaryAsync(Document document, Vari
         }, cancellationToken).ConfigureAwait(false);
 
         return document;
+
+        SyntaxNode RewriteScope(SyntaxNode scope)
+        {
+            var editor = new SyntaxEditor(scope, document.Project.Solution.Services);
+
+            foreach (var identifier in conflictReferences)
+                editor.ReplaceNode(identifier, identifier.WithAdditionalAnnotations(CreateConflictAnnotation()));
+
+            foreach (var identifier in nonConflictReferences)
+            {
+                if (identifier.Parent is AnonymousObjectMemberDeclaratorSyntax { NameEquals: null } anonymousMember)
+                {
+                    // Special case inlining into anonymous types to ensure that we keep property names:
+                    //
+                    // E.g.
+                    //     int x = 42;
+                    //     var a = new { x; };
+                    //
+                    // Should become:
+                    //     var a = new { x = 42; };
+                    editor.ReplaceNode(
+                        anonymousMember,
+                        anonymousMember.Update(NameEquals(identifier), expressionToInline).WithTriviaFrom(anonymousMember));
+                }
+                else if (identifier.Parent is ArgumentSyntax
+                {
+                    Parent: TupleExpressionSyntax tupleExpression,
+                    NameColon: null,
+                } argument &&
+                    !SyntaxFacts.IsReservedTupleElementName(identifier.Identifier.ValueText) &&
+                    tupleExpression.Arguments.Count(a => nonConflictReferences.Contains(a.Expression)) == 1)
+                {
```

Comment:
Indentation seems off here.
```suggestion
                else if (identifier.Parent is ArgumentSyntax
                    {
                        Parent: TupleExpressionSyntax tupleExpression,
                        NameColon: null,
                    } argument &&
                    !SyntaxFacts.IsReservedTupleElementName(identifier.Identifier.ValueText) &&
                    tupleExpression.Arguments.Count(a => nonConflictReferences.Contains(a.Expression)) == 1)
                {
```

Reply:
The Formatter doesn't like that, and gives an ide0055 error :+)